oap-logstream-data-object/src/main/java/oap/logstream/data/object/BinaryObjectLogger.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) Open Application Platform Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package oap.logstream.data.object;
+
+import com.google.common.base.Preconditions;
+import oap.dictionary.Dictionary;
+import oap.dictionary.DictionaryRoot;
+import oap.logstream.AbstractLoggerBackend;
+import oap.logstream.AvailabilityReport;
+import oap.net.Inet;
+import oap.reflect.TypeRef;
+import oap.template.DataModelTemplate;
+import oap.template.Template;
+import oap.template.TemplateAccumulatorBinary;
+import oap.template.TemplateEngine;
+import oap.template.TemplateException;
+import oap.template.Types;
+import oap.util.FastByteArrayOutputStream;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.annotation.Nonnull;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+import static oap.template.ErrorStrategy.ERROR;
+
+public class BinaryObjectLogger extends DataModelTemplate {
+    public final AbstractLoggerBackend backend;
+    public boolean typeValidation = true;

```

Comment:
Should it be configured? Is it the same as in `ObjectLogModel.java`

Reply:
ObjectLogModel deleted

Reply:
I mean `Should typeValidation be configured?` in dsp-kubernetes project? Or it should be always `true`?